lib/ansible/modules/packaging/os/package_facts.py

```diff
@@ -139,95 +150,256 @@
           }
         }
 '''
+import json
+
+from abc import ABCMeta, abstractmethod
+
+from ansible.module_utils.basic import AnsibleModule, get_all_subclasses
+from ansible.module_utils._text import to_native, to_text
+
+
+class TestFailed(Exception):
+    pass
+
+
+class PkgMgr(object):
+
+    __metaclass__ = ABCMeta
```

Comment:
The correct way to use metaclasses in code that needs to run on both python2 and python3 is:
``` python
from abc import ABCMeta
from six import with_metaclass


class PkgMgr(with_metaclass(ABCMeta, object)):
```